This failed for reasons I'm having a hard time correlating to the PR. Maybe we found flakiness in the backup tests?

```
-----> checking if database is running
pg_ctl: server is running (PID: 1)
/usr/lib/postgresql/9.4/bin/postgres
-----> checking if minio has 5 backups
!!!    did not find 5 base backups, which is the default (found 6)
make: *** [test] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -3,15 +3,10 @@
 # - Docker image name
 # - Kubernetes service, rc, pod, secret, volume names
 SHORT_NAME := postgres
-
-VERSION ?= git-$(shell git rev-parse --short HEAD)
-
-# Legacy support for DEV_REGISTRY, plus new support for DEIS_REGISTRY.
 DEIS_REGISTY ?= ${DEV_REGISTRY}/
 IMAGE_PREFIX ?= deis
 
-# Canonical docker image name
-IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${VERSION}
+include versioning.mk
 
 all: docker-build docker-push
 
@@ -19,10 +14,7 @@ all: docker-build docker-push
 # We also alter the RC file to set the image name.
 docker-build:
 	docker build --rm -t ${IMAGE} rootfs
-
-# Push to a registry that Kubernetes can access.
-docker-push:
-	docker push ${IMAGE}
+	docker tag -f ${IMAGE} ${MUTABLE_IMAGE}
 
 test:
 	contrib/ci/test.sh ${IMAGE}

--- a/_scripts/deploy.sh
+++ b/_scripts/deploy.sh
@@ -5,7 +5,7 @@
 
 cd "$(dirname "$0")" || exit 1
 
-export IMAGE_PREFIX=deisci VERSION=v2-beta
+export IMAGE_PREFIX=deisci
 docker login -e="$DOCKER_EMAIL" -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
 DEIS_REGISTRY='' make -C .. docker-build docker-push
 docker login -e="$QUAY_EMAIL" -u="$QUAY_USERNAME" -p="$QUAY_PASSWORD" quay.io

--- a/versioning.mk
+++ b/versioning.mk
@@ -1,0 +1,22 @@
+MUTABLE_VERSION ?= canary
+VERSION ?= git-$(shell git rev-parse --short HEAD)
+
+IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${VERSION}
+MUTABLE_IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${MUTABLE_VERSION}
+
+info:
+	@echo "Build tag:       ${VERSION}"
+	@echo "Registry:        ${DEIS_REGISTRY}"
+	@echo "Immutable tag:   ${IMAGE}"
+	@echo "Mutable tag:     ${MUTABLE_IMAGE}"
+
+.PHONY: docker-push
+docker-push: docker-immutable-push docker-mutable-push
+
+.PHONY: docker-immutable-push
+docker-immutable-push:
+	docker push ${IMAGE}
+
+.PHONY: docker-mutable-push
+docker-mutable-push:
+	docker push ${MUTABLE_IMAGE}

